src/test/groovy/org/whaka/mock/EventCollectorTest.groovy

```diff
@@ -0,0 +1,84 @@
+package org.whaka.mock
+
+import spock.lang.Specification
+
+import java.util.function.BiConsumer
+import java.util.function.Predicate
+
+/**
+ * @author gdzabaev
+ */
+class EventCollectorTest extends Specification {
+
+    private BiConsumer<Listener, Integer> method = {l,e -> l.event(e)}
+    private Predicate<Integer> filter = Mock()
+    private EventCollector.EventHandler<Integer> filterEH = Mock()
+    private EventCollector<Listener, Integer> collector = EventCollector.create(Listener.class, method, filter, filterEH)
+    private Listener l = collector.getTarget()
+
+    def "all filters are called" () {
+        when:
+            l.event(event)
+            l.event(event * 2)
+        then:
+            1 * filter.test(event) >> true
+            1 * filterEH.test(event) >> true
+        and:
+            1 * filter.test(event * 2) >> false
+            1 * filterEH.test(event * 2) >> false
+        where:
+            event << 100
+    }
+
+    def "accepted is called" () {
+        when:
+            l.event(event)
+            l.event(event * 2)
+        then:
+            1 * filter.test(event) >> true
+            1 * filterEH.test(event) >> true
+        and:
+            1 * filterEH.eventCollected(event)
+
+        and:
+            1 * filter.test(event * 2) >> false
+            1 * filterEH.test(event * 2) >> true
+        and:
+            0 * filterEH.eventCollected(_)
+
+        where:
+            event << 100
+    }
```

Comment:
There're clearly two distinct checks present - it can be split in two `where-then-and` blocks:
```
when:
	l.event(event)
then:
	1 * filter.test(event) >> true
	1 * filterEH.test(event) >> true
and:
	1 * filterEH.eventCollected(event)

when:
	l.event(event * 2)
then:
	1 * filter.test(event * 2) >> false
	1 * filterEH.test(event * 2) >> true
and:
	0 * filterEH.eventCollected(_)
```